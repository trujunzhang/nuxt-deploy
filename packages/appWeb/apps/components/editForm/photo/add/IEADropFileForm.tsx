import * as React from 'react'
import Dropzone from '@appComponents/editForm/photo/add/dropzoneApi'
import * as Telescope from '@appComponents/index'

interface IIEADropFileFormProps {
  onBeforeDropHook: any
  onAfterDropHook: (file: any) => any
}

export class IEADropFileForm extends React.Component<IIEADropFileFormProps, {}> {
  onDrop(accepted, rejected) {
    if (accepted.length === 1) {
      const file = accepted[0]
      if (file) {
        this.props.onBeforeDropHook()
        const reader = new FileReader()
        reader.onloadend = () => {
          this.props.onAfterDropHook(file)
        }
        reader.readAsDataURL(file)
      }
    }
  }

  render(){
    return null;
  }

  // TODO: DJZHANG(2020)
  // render() {
  //   return (
  //     <div className="upload upload--photos js-html5-uploader hidden" style={{ display: 'block' }}>
  //       <Telescope.F8AppAlertSection />

  //       <Dropzone
  //         multiple={false}
  //         accept="image/jpeg, image/png"
  //         className="file-drop file-drop--big js-file-drop"
  //         onDrop={this.onDrop.bind(this)}>
  //         <div className="file-drop_area">
  //           <div className="file-drop_area-inner">
  //             <h1>{'Drag and drop your photos here'}</h1>

  //             <fieldset className="hr-line">
  //               <legend>OR</legend>
  //             </fieldset>

  //             <div className="file-browser js-file-browser">
  //               <button
  //                 type="submit"
  //                 value="submit"
  //                 className="ybtn ybtn--primary file-browser_button">
  //                 <span>{'Browse Files'}</span>
  //               </button>
  //             </div>
  //           </div>
  //         </div>
  //       </Dropzone>
  //     </div>
  //   )
  // }
}
