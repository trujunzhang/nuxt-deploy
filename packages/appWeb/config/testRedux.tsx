import * as React from 'react'
import Telescope from '@appComponents/index'
import { I18nextProvider } from '@app/tools'

// get language from query parameter or url path
const lang = 'en'

import { getLocalI18n } from '@app/tools'
// import * as resources from '@assets/locales/i18nextLocalesLoader'

// export const mockedI18n = getLocalI18n(resources, (callback: any) => {
//   return 'en'
// })


// Store Initialization
// ------------------------------------
import createStore from '@src/redux/create'
const store = createStore()

import { createStore as RDCreateStore, Store } from 'redux';
import { Provider } from 'react-redux'
import withRedux, { MakeStore } from 'next-redux-wrapper';

import reducer from '@src/redux/rootReducer'
export type RootState = ReturnType<typeof reducer>;
/**
 * @param initialState The store's initial state (on the client side, the state of the server-side store is passed here)
 */
const makeStore: MakeStore = (initialState: RootState) => {
  return RDCreateStore(reducer, initialState);
};

export const testRedux = (key, showHeaderPanel = true) => {
  class Common extends React.Component {
    constructor(props) {
      super(props)
    }

  //   render() {
  //     const comp = Telescope.components[key]
  //     const children = React.createElement(comp, {})
  //     return (
  //       <Provider store={store}>
  //         <I18nextProvider i18n={mockedI18n}>
  //           <Telescope.components.Layout showHeaderPanel={showHeaderPanel}>
  //             {children}
  //           </Telescope.components.Layout>
  //         </I18nextProvider>
  //       </Provider>
  //     )
  //   }
  }

  return withRedux(makeStore)(Common)
}
