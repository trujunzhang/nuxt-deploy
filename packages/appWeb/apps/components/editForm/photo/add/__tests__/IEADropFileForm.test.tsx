import * as React from 'react'

// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer'

import { IEADropFileForm } from '../IEADropFileForm'
import { testRedux } from '@config/testRedux'

describe('Button', () => {
  it('render correctly for IEADropFileForm', () => {
    const children = <IEADropFileForm onBeforeDropHook={() => {}} onAfterDropHook={() => {}} />
    const renderPage = testRedux(children)
    const instance = renderer.create(renderPage)

    const page = instance.toJSON()
    expect(page).toMatchSnapshot()
  })
})
