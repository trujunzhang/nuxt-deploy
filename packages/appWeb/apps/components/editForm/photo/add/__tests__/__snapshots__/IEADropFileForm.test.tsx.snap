// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly for IEADropFileForm 1`] = `
<div
  className="upload upload--photos js-html5-uploader hidden"
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    aria-disabled={false}
    className="file-drop file-drop--big js-file-drop"
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="file-drop_area"
    >
      <div
        className="file-drop_area-inner"
      >
        <h1>
          Drag and drop your photos here
        </h1>
        <fieldset
          className="hr-line"
        >
          <legend>
            OR
          </legend>
        </fieldset>
        <div
          className="file-browser js-file-browser"
        >
          <button
            className="ybtn ybtn--primary file-browser_button"
            type="submit"
            value="submit"
          >
            <span>
              Browse Files
            </span>
          </button>
        </div>
      </div>
    </div>
    <input
      accept="image/jpeg, image/png"
      autoComplete="off"
      disabled={false}
      multiple={false}
      onChange={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.00001,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      type="file"
    />
  </div>
</div>
`;
